lib/launcher.js

```diff
@@ -4,120 +4,78 @@
 
-  const overridesFolder = options.debugFrontend
-    ? path.dirname(require.resolve(`../front_end/${options.appName}.json`))
-    : path.dirname(path.join(options.releaseFrontendFolder, `${options.appName}.js`));
+  process.title = 'ndb/main';
+  const appName = 'ndb_app';
+  const debugFrontend = !!process.env.NDB_DEBUG_FRONTEND;
 
+  const overridesFolder = debugFrontend
+    ? path.dirname(require.resolve(`../front_end/${appName}.json`))
+    : path.join(__dirname, '..', '.local-frontend');
   app.serveFolder(overridesFolder);
   app.serveFolder(path.join(__dirname, '..'));
-  if (options.debugFrontend) {
+  if (debugFrontend) {
     try {
       app.serveFolder(path.dirname(require.resolve(`chrome-devtools-frontend/front_end/ndb_app.json`)));
     } catch (e) {
       console.log('To use NDB_DEBUG_FRONTEND=1 you should run npm install from ndb folder first');
       process.exit(1);
     }
   }
-  process.title = 'ndb/main';
-  const backend = new Backend(app);
+  const backends = new Set();
   app.on('exit', () => {
     backend.dispose();
     setTimeout(() => process.exit(0), 0);
   });
   const params = [
     ['experiments', true],
-    ['debugFrontend', options.debugFrontend],
+    ['debugFrontend', debugFrontend],
     ['sources.hide_add_folder', true],
-    ['sources.hide_thread_sidebar', true]];
+    ['sources.hide_thread_sidebar', true],
+    ['panel', 'sources']];
   const paramString = params.reduce((acc, p) => acc += `${p[0]}=${p[1]}&`, '');
-  await app.load(`${options.appName}.html?${paramString}`, rpc.handle(backend));
-  return app;
-}
-
-async function setupUserDataDir(configDirectory, doNotCopyPreferences) {
-  if (!fs.existsSync(configDirectory))
-    await fsMkdir(configDirectory);
-
-  if (!fs.existsSync(path.join(configDirectory, 'ndd_store')))
-    await fsMkdir(path.join(configDirectory, 'ndd_store'));
-
-  const chromiumPreferencesFile = path.join(configDirectory, 'ChromiumPreferences');
-  if (!fs.existsSync(chromiumPreferencesFile))
-    await fsWriteFile(chromiumPreferencesFile, '{}', 'utf8');
-
-  const userDataDir = await fsMkdtemp(path.join(os.tmpdir(), 'ndb-'));
-  const defaultUserDir = path.join(userDataDir, 'Default');
-  await fsMkdir(defaultUserDir);
-
-  const preferencesFile = path.join(defaultUserDir, 'Preferences');
-  await fsCopyFile(chromiumPreferencesFile, preferencesFile);
-
-  process.on('exit', _ => {
-    if (!doNotCopyPreferences) {
-      if (fs.copyFileSync)
-        fs.copyFileSync(preferencesFile, chromiumPreferencesFile);
-      else
-        require('fs-copy-file-sync')(preferencesFile, chromiumPreferencesFile);
-    }
-    // rimraf might fail if Chrome is writing write now.
-    for (let i = 0; i < 5; ++i) {
-      try {
-        removeFolder.sync(userDataDir);
-        break;
-      } catch (e) {
-      }
-    }
+  app.on('window', async window => {
+    const backend = new Backend(window);
+    backends.add(backend);
+    const info = await window.evaluate(() => document.title);
```

Comment:
I'd only pass front_end params via url, the rest shoud go through backend.

Reply:
done!